app/assets/javascripts/discourse/app/components/modal/flag.js

```diff
@@ -100,6 +101,10 @@ export default class Flag extends Component {
       return true;
     }
 
+    if (this.selected.isIllegal && !this.isConfirmed) {
```

Comment:
I think this is fine for now, as you say it's very unlikely that this will be reused...